lineman/app/components/change_picture_form/change_picture_form.coffee

```diff
@@ -0,0 +1,19 @@
+angular.module('loomioApp').factory 'ChangePictureForm', ->
+  templateUrl: 'generated/components/change_picture_form/change_picture_form.html'
+  controller: ($scope, $timeout, CurrentUser, Records, FlashService) ->
+    $scope.user = CurrentUser.clone()
+
+    $scope.submit = (kind) ->
+      $scope.user.avatarKind = kind
+      Records.users.updateProfile($scope.user).then changePictureSuccess
+
+    $scope.selectFile = ->
+      $timeout -> angular.element('.change-picture-form__file-input').click()
```

Comment:
We define this behaviour in the link function of the attachment_form directive (the only other place we do file uploads at the moment), but I couldn't figure out how to add a link function to a modal. This is mildly hacky, but works fine.